pkg/controllers/work/apply_controller.go

```diff
@@ -629,6 +672,144 @@ func buildManifestAppliedCondition(err error, action applyAction, observedGenera
 	}
 }
 
+func (r *ApplyWorkReconciler) createConfigMap(ctx context.Context, configMapName string, owner metav1.OwnerReference, manifestHash, lastModifiedConfig string) error {
+	configMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: fleetSystemNamespace,
+		},
+		Data: map[string]string{},
+	}
+	configMap.Data[manifestHashKey] = manifestHash
+	configMap.Data[lastAppliedConfigKey] = lastModifiedConfig
+	// add applied work owner reference. let's check if it works
+	addOwnerRef(owner, configMap)
+	if err := r.spokeClient.Create(ctx, configMap); err != nil {
+		klog.ErrorS(err, "config map create failed", "configMap")
+		return err
+	}
+	klog.V(2).InfoS("config map was created", "configMap", configMap.GetName())
+	return nil
+}
+
+func (r *ApplyWorkReconciler) getConfigMap(ctx context.Context, obj *unstructured.Unstructured, owner metav1.OwnerReference) (*v1.ConfigMap, error) {
+	var configMap v1.ConfigMap
+	annotations := obj.GetAnnotations()
+	configMapName, ok := annotations[configMapNameAnnotation]
+	if !ok {
+		return nil, errors.New("manifest doesn't have a config map")
+	}
+	if err := r.spokeClient.Get(ctx, types.NamespacedName{Name: configMapName, Namespace: fleetSystemNamespace}, &configMap); err != nil {
```

Comment:
Hi Arvind! I understand the purpose of this code snippet here, but it could really use some refactoring.

* The createConfigMap call happens for the third time
* It really is not a very good idea to use recursion here.